.github/workflows/black_and_test.yml

```diff
@@ -1,4 +1,4 @@
-name: Black
+name: Black and Test
```

Comment:
I would add another _workflow_ for tests instead of running them together with style check.

The rationale for that is that it would make feedback more straight forward: collaborators can know whether they've broken something (tests fail) or if it's just a matter of style (style checks fail)  without even looking in the CI logs.

[This repository](https://github.com/cuducos/bot-followers) might be an example of two different workflows for test and style check.